### DISCUSSION TITLE
Micropython for ESP32-C3 network module missing

### DISCUSSION THREAD

**Discussion by BOB63** [micropython/micropython#15384] [ESP32] [answered]
Hi , I've seen that in the last two release Micropython firmware is missing the network module ( I've not cheked in older release) , that is many time included into sketch where is required wifi connection. Is it substituted with some other module or what ?
Is there any way to download the module if isn't part of firmware ? 
I've seen in google talk of mpip , upip but i've seen that in the modules available is present only pip , but has not the attribute install so pip.install("pakage_name") dosen't work.
Tks. Roberto

**Comment by glenn20** [comment]
The network module seems to be present on micropython 1.22 and 1.23 firmwares, eg. the network module is included in the images from the download page (eg. https://micropython.org/resources/firmware/ESP32_GENERIC-20240602-v1.23.0.bin):

```py
import network
```
Can you provide more information, ie. are you compiling your own images or using downloaded images? Which device and which firmware image are you using? How do you know that the `network` module is not included in the firmware?

**Reply by BOB63** [reply]
Hi @glenn20 ,
I've downloaded these two versions ESP32_GENERIC_C3-20231005-v1.21.0 and ESP32_GENERIC_C3-20240602-v1.23.0 because I'm using an ESP32-c3 mino board. inboth case Network module isn't present .
I've downloaded the generic version suggeste by you but seems that dosen't work with ESP32-c3.

_F:\>esptool --chip esp32c3 --port com8 --baud 460800 write_flash -z 0x0 F:\SWMIO\Elettronica\ESP32-C6\Micropython_esp32c3\ESP32_GENERIC-20240602-v1.23.0.bin
esptool.py v4.7.0
Serial port com8
Connecting...
Failed to get PID of a device on com8, using standard reset sequence.

Chip is ESP32-C3 (QFN32) (revision v0.4)
Features: WiFi, BLE, Embedded Flash 4MB (XMC)
Crystal is 40MHz
MAC: ec:da:3b:bd:d5:d8
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 460800
Changed.
Configuring flash size...
Unexpected chip id in image. Expected 5 but value was 0. Is this image for a different chip model?_

**A fatal error occurred: F:\SWMIO\Elettronica\ESP32-C6\Micropython_esp32c3\ESP32_GENERIC-20240602-v1.23.0.bin is not an ESP32-C3 image. Use --force to flash anyway.**

NOTE : Also forcing the upload I can't import the network module.

**Reply by mattytrentini** [reply]
I _think_ you may be trying to deploy an incorrect file - note the missing "_C3" in the filename you're passing to the `esptool` command.

**Reply by BOB63** [reply]
Hi @mattytrentini , yes infact it dosen't works. The problems is that using  the specific release for ESP32-C3  ( see initial post ) the module network seems to be not present.

**Reply by glenn20** [reply]
Ah. I'm afraid I dont have a c3 device, so cant confirm the presence of the network module.

Yes, you should not try to flash the `ESP32_GENERIC-20240602-v1.23.0.bin` image onto a C3 device. I only mentioned that because I couldnt see what device you were using in your message (although now I can see that you mentioned it in the title - an oversight on my part).

**Reply by robert-hh** [reply]
What is the output at REPL of the command:

`print(help("modules"))`

**Reply by BOB63** [reply, marked as answer]
I'M A STUPID !!!!! 
I was using Thonny and forgot to update the interpreter for micropython esp32 so it was working with Python 3 !!!!!!
Changing the interpreter it works.   
Lesson learned : Never start to program after 2 AM  . :-(

**Reply by sosi-deadeye** [reply]
This happened more than once. If Thonny couldn't see any serial ports, it just uses the installed Python 3 interpreter.
